src/main/java/org/codehaus/plexus/components/io/attributes/FileAttributes.java

```diff
@@ -160,50 +165,59 @@ private boolean containsPermission( PosixFilePermission groupExecute )
         return permissions.contains( groupExecute );
     }
 
+    @Override
     public boolean isGroupReadable()
     {
         return containsPermission( PosixFilePermission.GROUP_READ );
     }
 
+    @Override
     public boolean isGroupWritable()
     {
         return containsPermission( PosixFilePermission.GROUP_WRITE );
     }
 
+    @Override
     public boolean isOwnerExecutable()
     {
         return containsPermission( PosixFilePermission.OWNER_EXECUTE );
     }
 
+    @Override
     public boolean isOwnerReadable()
     {
         return containsPermission( PosixFilePermission.OWNER_READ );
     }
 
+    @Override
     public boolean isOwnerWritable()
     {
         return containsPermission( PosixFilePermission.OWNER_WRITE );
     }
 
+    @Override
     public boolean isWorldExecutable()
     {
         return containsPermission( PosixFilePermission.OTHERS_EXECUTE );
 
     }
 
+    @Override
     public boolean isWorldReadable()
     {
         return containsPermission( PosixFilePermission.OTHERS_READ );
     }
 
+    @Override
     public boolean isWorldWritable()
     {
         return containsPermission( PosixFilePermission.OTHERS_WRITE );
     }
 
+    @Override
     public String toString()
     {
-        StringBuilder sb = new StringBuilder();
+        StringBuilder sb = new StringBuilder( 128 );
```

Comment:
Why 128?

Reply:
For a completely empty attributes (no user, group, uid, gid...) the string created is about 75 chars, the default capacity for a StringBuilder is 16, so a small help to pre-allocate the capacity would reduce the time to construct the string, anyway, 128 might be high as is expecting all the values full, so I could reduce it to 80 to be more conservative.